environs/networking.go

```diff
@@ -105,3 +117,19 @@ func SupportsContainerAddresses(env Environ) bool {
 	}
 	return ok
 }
+
+// ProviderSpaceInfo contains all the information about a space needed
+// by another environ to decide whether it can be routed to.
+type ProviderSpaceInfo struct {
+	network.SpaceInfo
+
+	// This identifies the cloud containing the space. Care must be
```

Comment:
In a hangout Andrew pointed out that having a CloudSpec that wasn't quite a real CloudSpec was likely to bite us at some point (by some provider not scrubbing secrets and just using its real CloudSpec, for example). Changed the `ProviderSpaceInfo` just to have `CloudType` and everything else will live inside `ProviderAttributes`.